src/rules/ResourceSourceICUUnexplainedParams.js

```diff
@@ -0,0 +1,159 @@
+/*
+ * ResourceSourceICUUnexplainedParams.js
+ *
+ * Copyright © 2023 JEDLSoft
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Result } from "i18nlint-common";
+import { Resource } from "ilib-tools-common";
+import ResourceRule from "./ResourceRule.js";
+import Locale from "ilib-locale";
+import { IntlMessageFormat } from "intl-messageformat";
+import log4js from "log4js";
+
+// type imports
+/** @ignore @typedef {import("@formatjs/icu-messageformat-parser").MessageFormatElement} MessageFormatElement */
+/** @ignore @typedef {import("@formatjs/icu-messageformat-parser").ArgumentElement} ArgumentElement */
+/** @ignore @typedef {import("./ResourceSourceICUPluralCategories.js").Location} Location */
+
+const logger = log4js.getLogger("i18nlint.ResourceSourceICUUnexplainedParams");
+
+/**
+ * Check if replacement parameters mentioned in the English source string are
+ * mentioned in the description field as well. Translators can do a much better
+ * job of translating if they know what each replacement parameter represents.
+ */
+export class ResourceSourceICUUnexplainedParams extends ResourceRule {
+    /** @override */
+    name = "source-unexplained-params";
+    /** @override */
+    description =
+        "Check if replacement parameters used in resource source string are mentioned in resource description.";
```

Comment:
-> "Check if replacement parameters used in the source string are explained in the comments for the translators."